### PR TITLE
Auth tests fail when credentials are not specified.

### PR DESCRIPTION
**Pull request by laliberte**
This PR makes the auth fail when the credential environment variables are not set.

The method used here tests whether a test_url gives a 200 response.

### CODE REVIEW

**Comment by laliberte** [comment]
@jameshiebert Any chance this PR could be merged? This would make it much easier to run the test suite...

**Comment by jameshiebert** [comment]
@laliberte Thanks for working on this. Sorry, I'm pretty booked up this week, but I can take a look at it first thing next week. And FYI, since your last commit just offsets the first commit, next time you can do a [squash rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits) and just simplify that into a single commit.

**Comment by laliberte** [comment]
Thanks @jameshiebert ! No worries, nothing's urgent!

I currently trying to port [netcdf4_pydap](github/laliberte/netcdf4_pydap) from ``pydap`` version ``3.1.1`` to ``3.2.0``. I will submit a couple more PRs in that attempt.

Sorry for the numerous commits. I'm still trying to figure out the proper way to contributing to github!

**Comment by laliberte** [comment]
@jameshiebert 
Could you look at this PR? It would be very convenient to have working auth tests in the master branch.

**Comment by laliberte** [comment]
@jameshiebert Could we merge this and #49? These are minor and concern authentications. I've just rebased this PR and it should merge without problems.

**Comment by jameshiebert** [comment]
Everything looks pretty good except that I still get two tests which *error* if the credentials are not set. Like I've mentioned before, I would prefer if the tests fail (or get skipped)... but they should never error.

```
======================================================================
ERROR: test_dimension_esgf_query (pydap.tests.test_cas_esgf.TestESGF)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/hiebert/code/pydap/src/pydap/tests/test_cas_esgf.py", line 66, in test_dimension_esgf_query
    res = pydap.net.follow_redirect(self.test_url, session=session)
  File "/home/hiebert/code/pydap/src/pydap/net.py", line 70, in follow_redirect
    cookies_dict=cookies_dict)
  File "/home/hiebert/code/pydap/src/pydap/net.py", line 57, in follow_redirect
    res = req.get_response(application)
  File "/home/hiebert/code/pydap/freshenv/lib/python3.5/site-packages/webob/request.py", line 1316, in send
    application, catch_exc_info=False)
  File "/home/hiebert/code/pydap/freshenv/lib/python3.5/site-packages/webob/request.py", line 1280, in call_application
    app_iter = application(self.environ, start_response)
  File "/home/hiebert/code/pydap/freshenv/lib/python3.5/site-packages/webob/client.py", line 94, in __call__
    path, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1106, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1151, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/lib/python3.5/http/client.py", line 1260, in connect
    server_hostname=server_hostname)
  File "/usr/lib/python3.5/ssl.py", line 377, in wrap_socket
    _context=self)
  File "/usr/lib/python3.5/ssl.py", line 752, in __init__
    self.do_handshake()
  File "/usr/lib/python3.5/ssl.py", line 988, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.5/ssl.py", line 633, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)

======================================================================
ERROR: test_variable_esgf_query (pydap.tests.test_cas_esgf.TestESGF)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/hiebert/code/pydap/src/pydap/tests/test_cas_esgf.py", line 80, in test_variable_esgf_query
    res = pydap.net.follow_redirect(self.test_url, session=session)
  File "/home/hiebert/code/pydap/src/pydap/net.py", line 70, in follow_redirect
    cookies_dict=cookies_dict)
  File "/home/hiebert/code/pydap/src/pydap/net.py", line 57, in follow_redirect
    res = req.get_response(application)
  File "/home/hiebert/code/pydap/freshenv/lib/python3.5/site-packages/webob/request.py", line 1316, in send
    application, catch_exc_info=False)
  File "/home/hiebert/code/pydap/freshenv/lib/python3.5/site-packages/webob/request.py", line 1280, in call_application
    app_iter = application(self.environ, start_response)
  File "/home/hiebert/code/pydap/freshenv/lib/python3.5/site-packages/webob/client.py", line 94, in __call__
    path, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1106, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python3.5/http/client.py", line 1151, in _send_request
    self.endheaders(body)
  File "/usr/lib/python3.5/http/client.py", line 1102, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python3.5/http/client.py", line 934, in _send_output
    self.send(msg)
  File "/usr/lib/python3.5/http/client.py", line 877, in send
    self.connect()
  File "/usr/lib/python3.5/http/client.py", line 1260, in connect
    server_hostname=server_hostname)
  File "/usr/lib/python3.5/ssl.py", line 377, in wrap_socket
    _context=self)
  File "/usr/lib/python3.5/ssl.py", line 752, in __init__
    self.do_handshake()
  File "/usr/lib/python3.5/ssl.py", line 988, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib/python3.5/ssl.py", line 633, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:645)
```

**Comment by laliberte** [comment]
Darn! I thought that was fixed with my earlier fixes. This one should do the trick!

File: src/pydap/tests/test_cas_esgf.py [hunk 1]
```diff
@@ -33,22 +33,27 @@ def test_registration_esgf_auth(self):
...
+        assert(os.environ.get('PASSWORD_ESGF_NO_REG') is not None)
         with self.assertRaises(UserWarning):
             esgf.setup_session(os.environ.get('OPENID_ESGF_NO_REG'),
                                os.environ.get('PASSWORD_ESGF_NO_REG'),
                                check_url=self.url)
 
     def test_basic_esgf_auth(self):
         """
-        Set up PyDAP to use the URS request() function.
+        Set up PyDAP to use the ESGF request() function.
```
Comment (jameshiebert) [inline src/pydap/tests/test_cas_esgf.py]: Good catch

Comment (laliberte) [inline src/pydap/tests/test_cas_esgf.py]: :)

**Comment by jameshiebert** [comment]
In the assertions, I think that you may want to simply test for truthiness (e.g. ``assert os.environ.get('OPENID_URS')``), rather than asserting ``is not None``. In cases where the environment variable is set, but to an empty string (e.g. ``export OPENID_URS=``) the tests will continue to error out. However, None and the empty string are both non-truthy, so a simple assertion will do the right thing. It's not a huge deal, so I'll go ahead and merge this, but if you feel like following up with another PR, feel free to do so.

**Comment by laliberte** [comment]
Ok, will do now.

**Pull request merged**